### PR TITLE
Use check on condition number rather than smallest eigenvalue

### PR DESCRIPTION
**Pull request by mscroggs**

### CODE REVIEW

File: cpp/basix/math.cpp [hunk 1]
```diff
@@ -86,6 +94,34 @@ basix::math::eigh(const std::span<const double>& A, std::size_t n)
...
+double basix::math::rcond(const std::span<const double>& A, std::size_t n)
+{
+  // Copy A
+  std::vector<double> M(A.begin(), A.end());
+
+  int N = n;
+  char norm = '1'; // Compute the 1-norm
+  int ldA = n;
+  int info;
+  std::vector<double> work(4 * n);
```
Comment (garth-wells) [inline cpp/basix/math.cpp:106]: It would be better to requires the caller to pass in these work buffers.

File: cpp/basix/math.h [hunk 2]
```diff
@@ -68,6 +68,14 @@ std::array<typename U::value_type, 3> cross(const U& u, const V& v)
 std::pair<std::vector<double>, std::vector<double>>
 eigh(const std::span<const double>& A, std::size_t n);
 
+/// For a square matrix A, compute the norm of A multiplied by the reciprocal of
```
Comment (garth-wells) [inline cpp/basix/math.h:71]:
```suggestion
/// For a square matrix A, compute the norm of A multiplied by the (estimated) reciprocal of
```

Comment (garth-wells) [inline cpp/basix/math.h:71]: Which norm?

File: cpp/basix/precompute.cpp [hunk 3]
```diff
@@ -40,6 +40,10 @@ precompute::prepare_matrix(
   mdarray2_t permuted_matrix(dim, dim);
   std::vector<T> diag(dim);
 
+  const double rcond_matrix = math::rcond(
```
Comment (garth-wells) [inline cpp/basix/precompute.cpp:43]: Need a check here that the storage of `matrix` is contiguous, and check that it is square.

**Review by garth-wells — CHANGES_REQUESTED** [review]
What speed-up does the change give?

**Comment by mscroggs** [comment]
Closing this as #572 is a much better option

**Pull request closed without merging**